soh/soh/OTRGlobals.cpp

```diff
@@ -715,7 +715,26 @@ extern "C" void OTRExtScanner() {
     }
 }
 
+bool is_subpath(const std::filesystem::path& path, const std::filesystem::path& base) {
+    auto rel = std::filesystem::relative(path, base);
+    return !rel.empty() && rel.native()[0] != '.';
+}
+
 extern "C" void InitOTR() {
+#ifdef _WIN32
+    auto tempPath = std::filesystem::canonical(getenv("TEMP"));
```

Comment:
What happens if the `TEMP` env var doesn't exist, or if it is an illegal value that can't be parsed as a path?

We should probably have extra checks to avoid an exception, or consider a try/catch around all this so it doesn't crash.

Reply:
Yeah, I'll have a look. I'm pretty sure it just creates a blank or root path object, and the comparison just returns false, but I'm not certain. However, the likelihood of either of those happening, considering Windows sets that path itself on install, is pretty low.

Reply:
Alright, turns out that did crash, so I put in a check for it and recreated the default temp path by using `getenv("USERPROFILE")` and appending "AppData\Local\temp" to it. While running Windows without `TEMP` may technically be possible, if `USERPROFILE` doesn't exist, Windows won't be running, so I figure that's a save backup.